Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -83,7 +73,8 @@ struct TestOperationDescription: Decodable {
     // swiftlint:disable cyclomatic_complexity
 
     /// Runs the operation and asserts its results meet the expectation.
-    func validateExecution(
+    func validateExecution<T: SpecTest>(
+        test: inout T,
```

Comment:
why is `test` an inout parameter?

Reply:
basically just this question and otherwise first commit LGTM, nice to not have to validate the target type in each individual op

Reply:
it's `inout` because the targeted failpoint stuff updates state. if it wasn't `inout`, we'd get COW when the failpoints were triggered, so they'd never be disabled

Reply:
👍 comment might be useful for future readers?

Reply:
Added a commit to the end of this PR, and am just going to commit the first commit directly to avoid rebasing.

Reply:
sgtm

Reply:
I think I need to make lint changes to the commit anyways, so I'll just open a new PR to get the travis run and rebase this one afterwards.

Reply:
This PR has now been rebased off of master. The first commit (which contains the bulk of the diff) is just splitting up `TestOperation.swift` into many files. I can rebase off of that one too if you think the changes look good.

Reply:
the new file structure makes sense to me, let's go ahead and merge it 👍

Reply:
Opened #473 to run CI, will merge that once green.

Reply:
Ok, the split is merged and this PR is rebased on top. Diff should be much more readable now